src/pages/Top/index.tsx

```diff
@@ -1,26 +1,47 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
+import { useLocation } from 'react-router-dom'
 import { css } from '@emotion/react'
 import { colorToRgb } from '/@/utils/color'
 import { card } from '/@/utils/card'
+import api from '/@/utils/apis'
 
 import PlayerName from './PlayerName'
 import AvatarSelect from './AvatarSelect'
 
 // TODO:クエリがある場合は，招待リンクを踏んだパターンとして表示を変える
 // 招待リンク ?c=xxxxxroomIdxxxxx
 const Top = () => {
-  // TODO: アバター
+  const search = useLocation().search
+  const [roomId, setRoomId] = useState<string | null>(null)
+  const [hostName, setHostName] = useState<string | null>(null)
+
+  useEffect(() => {
+    const id = new URLSearchParams(search).get('c')
+    setRoomId(id)
+    // 招待リンクを踏んだ場合は，部屋ホストのユーザー名を取得する
+    if (id) {
```

Comment:
falsy な値はいるから `id !== null` か `id !== null || id.length === 0` のがいい鴨試練
```suggestion
    if (id !== null) {
```